client/html/themes/elegance/aimeos.js

```diff
@@ -4,3 +4,56 @@
  * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
  * @copyright Aimeos (aimeos.org), 2014
  */
+
+
+
+/**
+ * Hides or shows the catalog filter search reset button
+ * depending on if the input text field is empty or not.
+ */
+AimeosCatalogFilter.hideOrShowCatalogFilterTextSearchResetButton = function() {
+	if ($(this).val() !== '')
+		$(this.nextElementSibling).css('display', 'inline-block');
+	else
+		$(this.nextElementSibling).css('display', 'none');
+};
+
+
+
+/**
+ * Resets the currently catalog filter search.
+ * In the simplest case the entered search term was not submitted
+ * yet, such that the field is just cleared.
+ * The major purpose of this field is to reset the search without
+ * having to manually empty the input field and submit it to show
+ * all items of all categories again.
+ */
+AimeosCatalogFilter.resetCatalogFilterTextSearch = function() {
+//	var catalog_filter_search_input = $('.catalog-filter-search input:nth-of-type(1)');
+//	catalog_filter_search_input.val('');
+	$(this.previousElementSibling).val('');
+	this.parentNode.parentNode.submit();
```

Comment:
So testing with http://hamag-maschinenbau.de/index.php?id=13&L=43&ai[controller]=catalog&ai[action]=list&ai[f_search]=111&ai[f_name]=all&ai[f_catid]=1&cHash=10393b6a15c184e11f5e75a7fd6bef97
I finally agree. 
Especially as the Shop menu link already has the same effect.

Will push a new version of this branch without the submit of the form.